lib/routes/utils/oauth-mint.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const jwtool = require('fxa-jwtool')
+const Pool = require('../../pool')
+const rsa = require('../../crypto/rsa')
+
+// From the Firefox source code:
+// Having this parameter be short has limited security value and can cause
+// spurious authentication values if the client's clock is skewed and
+// we fail to adjust. See Bug 983256.
+const ASSERTION_LIFETIME = 1000 * 3600 * 24 * 365 * 25 // 25 years in ms.
+const CERT_LIFETIME = 1000 * 3600 * 5 // 5 hours in ms.
```

Comment:
I think actually the above doesn't apply to us.  We're talking server-to-server here, so we can expect well-synchronized clocks and there's some value in having the assertions be short-lived.  We could request e.g. a 5-minute lifetime on both cert and assertion here and have a reasonable expectation that it would all work OK.